core/domain/feedback_services_test.py

```diff
@@ -92,7 +87,7 @@ class SuggestionQueriesUnitTests(test_utils.GenericTestBase):
     CURRENT_TIME_IN_MSEC = 12345678
 
     def _generate_thread_id(self, unused_exp_id):
-        return self.THREAD_ID1
+        return self.EXP_ID2 + '.' + self.THREAD_ID1
```

Comment:
There's a lot of manual ID construction in this test. It doesn't feel like this generalizes to other types of suggestions.

If you want this to model an exploration suggestion, consider having a function somewhere whose job is to create the ID for suggestion threads pertaining to explorations, and call that function from this test. Don't hardcode the manual ID construction here, it risks causing the tests to skew from the main codebase.

Reply:
Actually these tests are going to be replaced in the new framework. So that's why I didn't try to generalize this test suite (the suggestions part of the feedback_services_test).

Reply:
I see -- I'd say generalize by referring to a specific function if you're planning to add that function in this PR. If not, then add it to the issue mentioned in https://github.com/oppia/oppia/pull/4971/files/7fa4895aa1a78a5e3a806cc7a6343fdc5888b714#r191315504 as a codebase deficiency that needs clean-up.

Prefer the former, because the codebase should always be clean -- no temporary deficiencies. But I can understand doing the latter if the former is hard for some reason.

Reply:
Now I'm not sure of what to do here. In the actual code we have a generate_thread_id function that generates the ID for the thread/suggestion. Here we are trying to mock that function (to remove the randomness in the generation of thread ids). So to have deterministic tests, it (kind of) makes sense to have this mock function return a hardcoded ID. I'll edit other instances of manual construction of IDs in this test suite.

Reply:
I removed the self.exp_id + '.' + thread_id lines and hardcoded the thread_ids at the start of the test suite.